src/main/dumpxdr.cpp

```diff
@@ -78,6 +78,43 @@ cereal_override(cereal::JSONOutputArchive& ar, const stellar::PublicKey& s,
 {
     xdr::archive(ar, stellar::KeyUtils::toStrKey<stellar::PublicKey>(s), field);
 }
+
+// A struct used to archive MuxedAccount
+// when type = KEY_TYPE_MUXED_ED25519.
+struct MuxedEd25519
```

Comment:
I don't understand the purpose of this type? Perhaps I'm missing something, but if you want to archive this information then why don't you just archive it directly from the relevant `cereal_overload`?

Reply:
`MuxedAccount` seems to have different levels of "nesting" based on the type.

For `KEY_TYPE_ED25519`, we print

```
...
              "sourceAccount": "GAY6WPTYIMNRRFEFOKUE3KRRKIMAWRB3XLSO2M6HNMIABBPDR5DTA3IY",
...
```

but, for `KEY_TYPE_MUXED_ED25519`, we print

```
...
                        "sourceAccount": {
                            "id": 0,
                            "accountID": "GAY6WPTYIMNRRFEFOKUE3KRRKIMAWRB3XLSO2M6HNMIABBPDR5DTA3IY"
                        },
                        "body": {
                            "type": "PAYMENT",
...
```

If I instead do `xdr::archive(ar, id, "id");` and `xdr::archive(ar, accountID, "accountID");`, it doesn't print "sourceAccount" and I end up with

```
...
                            "id": 0,
                            "accountID": "GAY6WPTYIMNRRFEFOKUE3KRRKIMAWRB3XLSO2M6HNMIABBPDR5DTA3IY",
                            "body": {
                                "type": "PAYMENT",
...
```

If I put the "id" and "accountID" in one string value and do `xdr::archive(ar, rendered_id_accountID, "sourceAccount")`, then I end with the multi-line printing issue because I won't know how much to indent.

This solution seems pretty weird, and I feel like I'm doing something very unnecessary, but I couldn't find other ways to do this.

Reply:
Removed